scripts/populate_expressions_list.py

```diff
@@ -0,0 +1,206 @@
+    text = text.replace('</th>','"')
+    text = text.replace('</tr>', '\n')
+    text = text.replace('</td><td>','", "')
+    text = text.replace('<td>','"')
+    text = text.replace('</td>','"')
+    text = text.replace('</thead>','')
+    text = text.replace('</table>','')
+    text = text.replace('<pre>','\n\n'+ filler)
+    text = text.replace('</pre>','\n\n')
+    text = re.sub(r"<a href='(.*)'>(.*)</a>", r"`\2 <\1>`_", text)
+
+    return text
+
+def format_function(function_dict):
+    f_name = function_dict['name']
+    if not 'variants' in function_dict:
+        # No variants make a list with a single function call
+        variants = [function_dict]
+    else:
+        # Multiple function calls
+        variants = function_dict['variants']
+
+    text = ''
+    for variant in variants:
+        # Format each function variant and add it to the function text
+        text += format_variant(variant, f_name)
+    return text
+
+def format_variant(function_dict, f_name):
+
+    # Organize arguments
+    arg_syntax_list = []
+    arg_description_list = []
+    optional_args = False
+    variable_args = ''
+    if 'arguments' in function_dict:
+        for arg in function_dict['arguments']:
+            if "optional" in arg and arg["optional"]:
+                if "default" in arg:
+                    default = f'={arg["default"]}'
+                else:
+                    default = ''
+                arg_text = f'[{arg["arg"]}{default}]'
+                optional_args = True
+            else:
+                arg_text = arg['arg']
+            if 'syntaxOnly' in arg:
+                arg_syntax_list.append(arg_text)
+                variable_args = ', ...'
+            elif 'descOnly' in arg:
+                arg_description_list.append(f"**{arg['arg']}** - {sphynxify_html(arg['description'],9)}")
+            else:
+                arg_syntax_list.append(arg_text)
+                arg_description_list.append(f"**{arg['arg']}** - {sphynxify_html(arg['description'],9)}")
+
+    # Prepare syntax and arguments strings
+    syntax = f'   * - Syntax\n     - '
+    if len(arg_syntax_list) > 0:
+        syntax += f'{f_name}({", ".join(arg_syntax_list)}{variable_args})\n'
+        descriptions = '\n       * '.join(arg_description_list)
+        arguments = (f"   * - Arguments\n"
+                     f"     - * {descriptions}\n"
+                     f"\n")
+    elif f_name.startswith("$"):
+        syntax += f"{f_name}\n"
+        arguments = ''
+    else:
+        syntax += f"{f_name}()\n"
+        arguments = ''
+
+    if optional_args:
+        syntax += '\n       [] marks optional arguments\n'
+
+    if 'type' in function_dict and function_dict['type'] == 'expression':
+        syntax = ''
+
+    # Prepare examples
+    if 'examples' in function_dict:
+        ex_list = [f"{ex['expression']} → {ex['returns']}" for ex in function_dict['examples']]
+        examples = "   * - Examples\n     - * "+ "\n\n       * ".join(ex_list)
```

Comment:
```suggestion
        ex_list = [(f"{ex['expression']} → {ex['returns']}" + (f"\n\n         {ex['note']}" if 'note' in ex else "")) for ex in function_dict['examples']]
        examples = "   * - Examples\n     - * " + "\n       * ".join(ex_list)
```
If we would like to include example notes (found in `clamp`, `scale_exp` and `scale_linear` - all in the *Math* group)